### PR TITLE
New polygon shape and additional drawing methods.

### PR DESCRIPTION
**Pull request by damonkohler**
Polygon shape uses triangluation to support concave polygons.
Adds lines to text shape to indicate the point it is associated with.

### CODE REVIEW

**Comment by damonkohler** [comment]
@SirVer PTAL

File: android_15/src/org/ros/android/view/visualization/shape/MetricSpacePolygon.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import javax.microedition.khronos.opengles.GL10;
+
+/**
+ * A polygon with metric space vertices.
+ *
+ * @author damonkohler@google.com (Damon Kohler)
+ */
+public class MetricSpacePolygon extends BaseShape {
+
```
Comment (SirVer) [inline android_15/src/org/ros/android/view/visualization/shape/MetricSpacePolygon.java:37]: empty line?

File: android_15/src/org/ros/android/view/visualization/shape/Triangulate.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    } else {
+      for (int v = 0; v < n; v++) {
+        V[v] = (n - 1) - v;
+      }
+    }
+
+    int nv = n;
+
+    // Remove nv-2 Vertices, creating 1 triangle every time.
+    int count = 2 * nv;   /* error detection */
```
Comment (SirVer) [inline android_15/src/org/ros/android/view/visualization/shape/Triangulate.java]: consistent commenting.

File: android_15/src/org/ros/android/view/visualization/shape/Triangulate.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      if (nv <= v) {
+        v = 0;  // new v
+      }
+      int w = v + 1;
+      if (nv <= w) {
+        w = 0;  // next
+      }
+
+      if (snip(contour, u, v, w, nv, V)) {
+        int a, b, c, s, t;
```
Comment (SirVer) [inline android_15/src/org/ros/android/view/visualization/shape/Triangulate.java]: declare where defined and final.

File: android_15/src/org/ros/android/view/visualization/shape/MetricSpacePolygon.java (hunk 1 above)
Comment (damonkohler) [inline android_15/src/org/ros/android/view/visualization/shape/MetricSpacePolygon.java:37]: That's normal.

File: android_15/src/org/ros/android/view/visualization/shape/Triangulate.java (hunk 2 above)
Comment (damonkohler) [inline android_15/src/org/ros/android/view/visualization/shape/Triangulate.java]: done

File: android_15/src/org/ros/android/view/visualization/shape/Triangulate.java (hunk 3 above)
Comment (damonkohler) [inline android_15/src/org/ros/android/view/visualization/shape/Triangulate.java]: done

**Pull request merged**